src/testFixtures/groovy/grails/plugin/geb/WebDriverContainerHolder.groovy

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2024 original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package grails.plugin.geb
+
+import geb.Browser
+import groovy.transform.EqualsAndHashCode
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.chrome.ChromeOptions
+import org.openqa.selenium.remote.RemoteWebDriver
+import org.spockframework.runtime.extension.IMethodInvocation
+import org.spockframework.runtime.model.SpecInfo
+import org.testcontainers.Testcontainers
+import org.testcontainers.containers.BrowserWebDriverContainer
+
+import java.time.Duration
+
+/**
+ * Responsible for initializing a {@link org.testcontainers.containers.BrowserWebDriverContainer BrowserWebDriverContainer}
+ * per the Spec's {@link grails.plugin.geb.ContainerGebConfiguration ContainerGebConfiguration}.  This class will try to
+ * reuse the same container if the configuration matches the current container.
+ *
+ * @author James Daugherty
+ * @since 5.0
+ */
+class WebDriverContainerHolder {
+
+    RecordingSettings recordingSettings
+    WebDriverContainerConfiguration configuration
+    BrowserWebDriverContainer currentContainer
+
+    WebDriverContainerHolder(RecordingSettings recordingSettings) {
+        this.recordingSettings = recordingSettings
+    }
+
+    boolean isInitialized() {
+        currentContainer != null
+    }
+
+    boolean stop() {
+        if (!currentContainer) {
+            return false
+        }
+        currentContainer.stop()
+        currentContainer = null
+        configuration = null
+        return true
+    }
+
+    boolean matchesCurrentContainerConfiguration(WebDriverContainerConfiguration specConfiguration) {
+        specConfiguration == configuration
+    }
+
+    private static int getPort(IMethodInvocation invocation) {
+        try {
+            return (int) invocation.instance.metaClass.getProperty(invocation.instance, 'serverPort')
+        } catch (Exception ignore) {
+            throw new IllegalStateException('Test class must be annotated with @Integration for serverPort to be injected')
+        }
+    }
+
+    boolean reinitialize(IMethodInvocation invocation) {
```

Comment:
camelCase?

Reply:
Are you proposing `reinitialize` become `reInitialize`?  

I would considered this one word.  Even the JVM seems to agree since `LinkedHashMap#reinitialize` exists

Reply:
You are correct 👍